android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java

```diff
@@ -263,7 +264,7 @@ public Completable resendEmailVerification(@NonNull final String email) {
 
                     if (!(throwable instanceof ConsentRequiredException)
                             || retryAttempt > 1
-                            || !isConsented()) {
+                            || isConsentedOnBridgeServer()) {
```

Comment:
These changes fix AA-8. There were two bugs here, !isConsented() was an incorrect conditional that was causing the error observable to always be returned and uploadConsentLocally() never was attempted.  And, the method "isConsented()" was inappropriate here since that call looks at local consent, and would always be considered consented in contrast to the ConsentRequiredException.

Reply:
Your logic makes sense to me when reading it, it may be possible that the changes I made were unnecessary.  I wasn't thinking in terms of "If there is no local consent, there is no reason to attempt another upload", I'll add that comment to "isConsented()", revert those methods changes, and see if it works.